multiapps-controller-core/src/main/java/org/cloudfoundry/multiapps/controller/core/cf/util/DeployedAfterModulesContentValidator.java

```diff
@@ -44,9 +56,20 @@ private boolean areModuleDependenciesResolvable(Module module, Set<String> modul
             return true;
         }
         return moduleDependencies.stream()
-                                 .filter(dependency -> !modulesInModelNames.contains(dependency))
-                                 .map(dependencyModule -> client.getApplication(dependencyModule, false))
-                                 .noneMatch(Objects::isNull);
+                                 .allMatch(dependency -> isModuleDependencyResolvable(modulesInModelNames, module.getName(), dependency));
+    }
+
+    private boolean isModuleDependencyResolvable(Set<String> modulesInModelNames, String moduleWithDependency, String dependencyModule) {
+        if (modulesInModelNames.contains(dependencyModule)) {
+            return true;
+        }
+        Module notContainedModule = allModulesInArchive.get(dependencyModule);
```

Comment:
What should be mean notContainedModule? I am asking this because if the module is contained in archive, so the module it contained.

Reply:
Because it is not contained in the list of modules for deployment. The if clause before that checks whether this module is specified for deployment (could not be in partial deploy scenario). Maybe a good decision is to rename `modulesInModelNames` to `modulesForDeployment` or something similar.

Reply:
I am agree with `modulesForDeployment` renaming. If you can to replace `notContainedModule` variable name to something which doesn't have `not` word it will be better. If you couldn't figure it out, leave it as it is.